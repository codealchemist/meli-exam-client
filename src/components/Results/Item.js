import React from 'react'
import { withRouter } from 'react-router-dom'
import { getProductDetails } from '../../api'
import Context from '../../Context'
import { Wrapper, LeftWrapper, RightWrapper, PriceRow } from './Elements'

const openDetails = ({ item, history, context }) => {
  const { id } = item
  console.log('Open Details', item)

  getProductDetails(id).then(response => {
    console.log('GOT product details response', response)
    context.setProductDetails(response)

    // Redirect to Details Page.
    history.push(`/items/${id}`)
  })
}

const Item = ({ item, history }) => (
  <Context.Consumer>
    {context => (
      <Wrapper onClick={() => openDetails({ item, history, context })}>
        <LeftWrapper>
          <img src={item.picture} alt="Product" />
        </LeftWrapper>

        <RightWrapper>
          <PriceRow>
            <h3>$ {item.price.amount}</h3>
            <label>{item.address.state_name}</label>
          </PriceRow>
          <p>{item.title}</p>
        </RightWrapper>
      </Wrapper>
    )}
  </Context.Consumer>
)

export default withRouter(Item)
