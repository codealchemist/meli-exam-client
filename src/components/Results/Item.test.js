import React from 'react'
import { mount } from 'enzyme'
import Item from './Item'

jest.mock('react-router-dom', () => ({
  withRouter: component => component
}))
jest.mock('../../Context', () => ({
  Consumer: obj =>
    obj.children({
      setProductDetails: jest.fn()
    })
}))

describe('Item', () => {
  const history = {
    push: jest.fn()
  }
  const item = {
    picture: 'pictureUrl',
    price: {
      amount: 42
    },
    title: 'Rock!',
    address: {
      state_name: 'BA'
    }
  }

  it('should render', () => {
    const wrapper = mount(<Item item={item} history={history} />)
    expect(wrapper).toMatchSnapshot()
  })
})
