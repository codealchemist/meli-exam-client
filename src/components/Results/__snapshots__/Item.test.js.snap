// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should render 1`] = `
<Item
  history={
    Object {
      "push": [MockFunction],
    }
  }
  item={
    Object {
      "address": Object {
        "state_name": "BA",
      },
      "picture": "pictureUrl",
      "price": Object {
        "amount": 42,
      },
      "title": "Rock!",
    }
  }
>
  <Consumer>
    <styled.div
      onClick={[Function]}
    >
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": true,
              "lastClassName": "hrDVpR",
              "rules": Array [
                "
  display: flex;
  position: relative;
  border-bottom: 1px solid #eee;
  padding: 10px 0;
  cursor: pointer;
  animation: all 2x;

  :hover {
    border-bottom: 1px solid ",
                "#fff159",
                ";
  }
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        onClick={[Function]}
      >
        <div
          className="sc-bdVaJa hrDVpR"
          onClick={[Function]}
        >
          <styled.div>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": true,
                    "lastClassName": "QtrCB",
                    "rules": Array [
                      "
  width: 160px;
  margin: auto;
  text-align: center;
  padding: 0 10px;

  img {
    border-radius: 4px;
  }
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="sc-bwzfXH QtrCB"
              >
                <img
                  alt="Product"
                  src="pictureUrl"
                />
              </div>
            </StyledComponent>
          </styled.div>
          <styled.div>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-htpNat",
                    "isStatic": true,
                    "lastClassName": "bzorVM",
                    "rules": Array [
                      "
  width: 100%;

  h3 {
    font-size: 24px;
    margin: 0;
    padding-right: 5px;
  }

  p {
    font-size: 20px;
  }
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-htpNat",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="sc-htpNat bzorVM"
              >
                <styled.div>
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-bxivhb",
                          "isStatic": true,
                          "lastClassName": "hhlAgH",
                          "rules": Array [
                            "
  display: flex;
  justify-content: space-between;
  align-items: center;

  label {
    padding: 0 10px;
  }
",
                          ],
                        },
                        "displayName": "styled.div",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-bxivhb",
                        "target": "div",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <div
                      className="sc-bxivhb hhlAgH"
                    >
                      <h3>
                        $ 
                        42
                      </h3>
                      <label>
                        BA
                      </label>
                    </div>
                  </StyledComponent>
                </styled.div>
                <p>
                  Rock!
                </p>
              </div>
            </StyledComponent>
          </styled.div>
        </div>
      </StyledComponent>
    </styled.div>
  </Consumer>
</Item>
`;
